subcommand/app-entrypoint/command_unix.go

```diff
@@ -0,0 +1,153 @@
+//go:build !windows
+// +build !windows
+
+package appentrypoint
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/consul-ecs/entrypoint"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+)
+
+const (
+	flagShutdownDelay = "shutdown-delay"
+)
+
+type Command struct {
+	UI      cli.Ui
+	log     hclog.Logger
+	once    sync.Once
+	flagSet *flag.FlagSet
+
+	sigs          chan os.Signal
+	appCmd        *entrypoint.Cmd
+	shutdownDelay time.Duration
+}
+
+func (c *Command) init() {
+	c.log = hclog.New(&hclog.LoggerOptions{Name: "consul-ecs"})
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.DurationVar(&c.shutdownDelay, flagShutdownDelay, 0,
+		`Continue running for this long after receiving SIGTERM. Must be a duration (e.g. "10s").`)
+	c.log = hclog.New(nil)
+
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+
+	// Flag parsing stops just before the first non-flag argument ("-" is a non-flag argument)
+	// or after the terminator "--"
+	err := c.flagSet.Parse(args)
+	if err != nil {
+		c.UI.Error(fmt.Sprint(err))
+		return 1
+	}
+
+	// Remaining args for the application command, after parsing our flags
+	args = c.flagSet.Args()
+
+	if len(args) == 0 {
+		c.UI.Error("command is required")
+		return 1
+	}
+
+	c.sigs = make(chan os.Signal, 1)
+	c.appCmd = entrypoint.NewCmd(c.log, args)
+
+	return c.realRun()
+}
+
+func (c *Command) realRun() int {
+	signal.Notify(c.sigs)
+	defer c.cleanup()
+
+	go c.appCmd.Run()
+	if _, ok := <-c.appCmd.Started(); !ok {
+		return 1
+	}
+
+	if exitCode, exited := c.waitForSigterm(); exited {
+		return exitCode
+	}
+	if c.shutdownDelay > 0 {
+		c.log.Info(fmt.Sprintf("consul-ecs: received sigterm. waiting %s before terminating application.", c.shutdownDelay))
```

Comment:
can we include the datetime here or use hclog? Might be useful in any debugging situaton

Reply:
This is using hclog, so we get a timestamp:

```
2021-11-30T10:49:05.150-0600 [INFO]  consul-ecs: received sigterm. waiting 5s before terminating application.
```

(Also, CloudWatch has its own timestamps)

Reply:
Ah nice, then can we use k/v logging instead of fmt.Sprintf?

Reply:
I guess for this log fmt.Sprintf reads better